google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/ChunkedDownloadCallable.java

```diff
@@ -57,17 +55,24 @@ final class ChunkedDownloadCallable implements Callable<DownloadSegment> {
   }
 
   @Override
-  public DownloadSegment call() throws Exception {
-    // TODO: Wrap Exceptions and put them in result
+  public DownloadSegment call() {
+    long bytesCopied = -1L;
     try (ReadChannel rc = storage.reader(originalBlob.getBlobId(), opts)) {
       FileChannel wc =
           FileChannel.open(destPath, StandardOpenOption.WRITE, StandardOpenOption.CREATE);
       rc.seek(startPosition);
       rc.limit(endPosition);
       wc.position(startPosition);
-      ByteStreams.copy(rc, wc);
-    } catch (IOException e) {
-      throw new StorageException(e);
+      bytesCopied = ByteStreams.copy(rc, wc);
```

Comment:
Since we know how many bytes we expect we should validate we receive them here.

There are circumstances where we've seen a connection get truncated without throwing an error, and are not retried.